src/compas_robots/viewer/scene/robotmodelobject.py

```diff
@@ -125,32 +129,33 @@ def transform(self, geometry, transformation: Transformation):
         """
         geometry.transformation = transformation * geometry.transformation
 
-    def create_geometry(self, geometry: Mesh, name: Optional[str] = None, color: Optional[Color] = None) -> MeshObject:
+    def create_geometry(self, item: Mesh, name: Optional[str] = None, color: Optional[Color] = None) -> MeshObject:
         """Create a mesh object from a given geometry.
 
         See Also
         --------
         :class:`compas_robots.scene.AbstractRobotModelObject`
         """
+        if not isinstance(item, Data):
+            raise ValueError("The item assigned to this scene object should be a data object: {}".format(type(item)))
         mesh_object = MeshObject(
-            geometry,
+            item=item,
             viewer=self.viewer,
             parent=None,
             is_selected=self.is_selected,
-            is_locked=self.is_locked,
-            is_visible=self.is_visible,
+            is_visible=self.show,
             show_points=self.show_points,
             show_lines=self.show_lines,
             show_faces=self.show_faces,
             facecolor=color,
-            lineswidth=self.lineswidth,
-            pointssize=self.pointssize,
+            linewidth=self.linewidth,
+            pointsize=self.pointsize,
             opacity=self.opacity,
             config=self.viewer.config,
             hide_coplanaredges=self.hide_coplanaredges,
             use_vertexcolors=self.use_vertexcolors,
             name=name,
-            context=self.scene.context,
+            context=self.context,
```

Comment:
99% of inputs here are already in `kwargs` of ___init___, you can save is as an attribute there then simply do something like 
```
mesh_object = MeshObject(
    item=item,
    **self.kwargs
)
```